xpk.py

```diff
@@ -1517,6 +1502,85 @@ def setup_docker_image(args) -> tuple[int, str]:
 
   return 0, docker_image
 
+def get_main_and_sidecar_container(args, system, docker_image, command) -> str:
+  """Generate yaml for main and sidecar container.
+  Args:
+    args: user provided arguments for running the command.
+    system: system characteristics
+    docker_image: docker image
+    command: command to run in the main container
+
+  Returns:
+    str:
+      yaml for main and sidecar container
+  """
+  yaml = """- name: stacktrace-explorer
+                image: busybox:1.28
+                args: [/bin/sh, -c, "while [ ! -d /tmp/debugging ]; do sleep 60; done; while [ ! -e /tmp/debugging/* ]; do sleep 60; done; tail -n+1 -f /tmp/debugging/*"]
+                volumeMounts:
+                - name: tpu-stack-trace
+                  readOnly: true
+                  mountPath: /tmp/debugging
+              - name: {args.docker_name}
```

Comment:
We want to avoid duplicating the yaml file (main container aspect) in two functions if possible, otherwise it can be harder to maintain and refactor changes to the main container in the future.

In your get_main_and_sidecar_container function, I recommend it directly calls the get_main_container function instead of replicating the main container details. LMK if you see any issues with this!